src/pictureNamingTask.ts

```diff
@@ -72,165 +64,157 @@ function getRandomElementWithSeed(array: any[]) {
     }
   } while (!foundUnique);
 
-  console.log("Selected Index:", randomIndex);
-  console.log("Selected Indices Set:", indiciesSelected);
-
-  let result = [array[randomIndex]];
-  console.log("***********************");
-  console.log("Result from random index:");
-  console.table(result);
-  console.log("***********************");
+  const result = [array[randomIndex]!];
   return result;
 }
 
 // draw an image at random from the bank depending on the difficulty_level selected
+// closure
 function createStimuli(
   difficultyLevel: number,
   language: string,
   clearSet: boolean,
-) {
+): ExperimentImage[] {
   if (clearSet === true) {
     indiciesSelected.clear();
   }
-  try {
-    let imgList = imageDB;
-    imgList = imgList.filter(
-      (image: any) =>
-        Number(image.difficultyLevel) === difficultyLevel &&
-        image.language === language,
-    );
-    console.log("filtered");
-    console.log(imgList);
-    let result = getRandomElementWithSeed(imgList);
-    return result;
-  } catch (error) {
-    console.error("Error fetching and parsing data:", error);
-  }
+  let imgList: ExperimentImage[] = imageDB;
+  imgList = imgList.filter(
```

Comment:
here, you can just set `imgList = imgDb.filter`

Reply:
see 8d21bb4